TreeDB/collections/column_asset_gc.go

```
@@ -469,6 +482,172 @@ func (c *Collection) columnAssetGC(ctx context.Context, opts ColumnAssetGCOption
+}
+
+func recoverableColumnAssetReplayCandidate(rootDir string, ref ColumnAssetRef, collection string, cfg ColumnStoreConfig, visibleRecoveryLSN uint64) (bool, error) {
+	if visibleRecoveryLSN == 0 {
+		return false, nil
+	}
+	raw, err := readColumnPhysicalAssetFromManager(rootDir, ref)
+	if err != nil {
+		// Exact namespace/identity validation immediately before deletion owns
+		// filesystem races. Bytes that cannot validate as this ref are not a
+		// replayable published asset and therefore do not gain a replay pin.
+		return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 624) but it returns nil

(nilerr)